### PR TITLE
Support markdown emoji

### PR DESCRIPTION
**Pull request by Yuuki77**
Fixes https://github.com/jesseduffield/lazygit/issues/1354

Added a flag called `parseEmoji`
if this flag true, the markdown in the commit message should be parsed.
Introduced github.com/kyokomi/emoji/v2 v2.2.8 to parse emoji.

but texts in Log are not parsed..
if you  think there is a better way to implement, please let me know 🙏 

also I thought if we support this library, we can just render it as markdown 🤔 
https://github.com/charmbracelet/glamour

image

### CODE REVIEW

File: pkg/gui/presentation/commits.go [hunk 1]
```diff
@@ -73,10 +74,18 @@ func getFullDescriptionDisplayStringsForCommit(c *models.Commit, cherryPickedCom
...
-	return []string{shaColor.Sprint(c.ShortSha()), secondColumnString, yellow.Sprint(truncatedAuthor), tagString + defaultColor.Sprint(c.Name)}
+	commitName := ""
+	if parseEmoji {
+		commitName = emoji.Sprintf(c.Name)
+	} else {
+		commitName = c.Name
+	}
+	// return []string{shaColor.Sprint(c.ShortSha()), secondColumnString, yellow.Sprint(truncatedAuthor), tagString + defaultColor.Sprint(c.Name)}
+
+	return []string{shaColor.Sprint(c.ShortSha()), secondColumnString, yellow.Sprint(truncatedAuthor), tagString + commitName}
```
Comment (mjarkk) [inline pkg/gui/presentation/commits.go:85]: Why did you remove `defaultColor.Sprint(..)`?

Comment (Yuuki77) [reply pkg/gui/presentation/commits.go]: @mjarkk thank you for the review!
sorry forgot to remove comment out 🙇 
fixed https://github.com/jesseduffield/lazygit/pull/1372/commits/eac9b2f6baee060b445633d2badf8fd8fa877710

**Review by mjarkk — APPROVED** [review]

**Comment by Sebclem** [comment]
Hi ! 
Any new on this being merged ?

**Comment by mjarkk** [comment]
Just merged in the latests master and the code seems to work fine.  
I'm not 100% comfortable merging this but as the default is false there should be no harm in doing so.

**Comment by mjarkk** [comment]
OH NO WHAT DID I DO

**Comment by mjarkk** [comment]
I'M SO SOO SORRY
I should not rebase in someone work
I'll try to recreate your changes

**Comment by mjarkk** [comment]
@Yuuki77 I've created #1387 to recreate your changes, would you like to review the changes and check if i missed something?  
And again sorry
